src/Aperture/elliptical.jl

```diff
@@ -0,0 +1,171 @@
+export EllipticalAperture,
+       EllipticalAnnulus
+
+"""
+   EllipticalAperture(x, y, a, b, theta)
+
+A elliptical aperture.
+
+Where a >= b > 0 and theta in degrees
+
+# Examples
+```jldoctest
+julia> EllipticalAperture(2,2,2,2,1.2)
+EllipticalAperture(2.0, 2.0, a=2.0, b=2.0, theta=1.2°)
+
+```
+"""
+struct EllipticalAperture{T <: Number} <: AbstractAperture
+    x::T
+    y::T
+    a::T
+    b::T
+    theta::T
+end
+
+function EllipticalAperture(x, y, a, b, theta)
+    if (b < 0.0 || a < b )
+            error("illegal ellipse parameters. Require a >= b > 0.0")
```

Comment:
`"Illegal..`

Reply:
I don't like the term "illegal". Perhaps use "invalid" instead.